src/seedu/addressbook/commands/Command.java

```diff
@@ -43,6 +43,13 @@ public CommandResult execute(){
         throw new UnsupportedOperationException("This method should be implement in child classes");
     }
 
+    /**
+     * Returns true if a Command mutates data.
+     */
+    public boolean isMutating(){
+        throw new UnsupportedOperationException("This method should be implement in child classes");
```

Comment:
The message "This method ..." should be stored in a string variable. Avoid magic numbers